### PR TITLE
feat: added a dropdown menu under avatar

### LINKED ISSUE #1031: Feature: menu/other UI from header avatar with username
On top bar, when hovering over the avatar it should show the username as text visible to see what username we have logged in.

### PR DESCRIPTION
**Pull request by blackgirlbytes**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This is my first pr to open source 🎉 

This PR adds a dropdown menu feature. 
Prior to this PR, when you clicked on your avatar in the top right-hand corner, it would immediately redirect you to your GitHub profile. Additionally, the logout button was on the navigation bar when logged in.

With the changes in this PR you now:

1. Click your avatar and get a dropdown menu with:  
- your github name where you will be redirected to your github profile if clicked
- A logout link
- The app version (I wasn't sure if I should do this because I know the admin nav may have these details, but the ticket suggested I add some info about the app. This is all the app info that I knew how to get)
2. Hover states:
- If you hover over the links, it will indicate that you hovered by turning a lighter shade of grey. This is not the case for the version number because it's not a link
 3. New component called Dropdown Menu added to storybook
 4. used useRef and useEffect hooks to ensure the dropdown menu closed when clicking away/outside of the component.

I am not sure if tests are needed because I have never written tests for React. Usually, I write tests on the backend.

## Related Tickets & Documents

Fixes #1031 
## Mobile & Desktop Screenshots/Recordings

Recording: https://www.loom.com/share/3b11d5b1b96445e992c754dd73fe2598
image
Screen Shot 2021-08-16 at 5 42 59 AM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
not that I know of

## [optional] What gif best describes this PR or how it makes you feel?
felicity

### CODE REVIEW

**Review by filiptronicek — CHANGES_REQUESTED** [review]
Hi @blackgirlbytes, I'm loving your approach to this feature. A job well done, it looks really well integrated into Open Sauced. I have a couple of things that may be worth checking out:

1. The element is not focusable (as far as I've tried) and so it may be harder to access without a mouse or a pointer.
2. The version could maybe be a hotlink for the GitHub tag in the repo.
3. The avatar element would probably benefit from a `pointer` cursor to indicate there is something the user can do with it.

**Comment by blackgirlbytes** [comment]
> Hi @blackgirlbytes, I'm loving your approach to this feature. A job well done, it looks really well integrated into Open Sauced. I have a couple of things that may be worth checking out:
> 
> 1. The element is not focusable (as far as I've tried) and so it may be harder to access without a mouse or a pointer.
> 2. The version could maybe be a hotlink for the GitHub tag in the repo.
> 3. The avatar element would probably benefit from a `pointer` cursor to indicate there is something the user can do with it.

@filiptronicek Thank you for the quick review!

1. I'm not familiar with making an element focusable by keyboard, so this is new for me. I tried adding tabIndex=0 to each element..and I was able to tab through it by keyboard. See recording below. If this is not what you meant, I may need extra assistance.
2. I added the hotlink to the GitHub tag. That was a great suggestion.
3. Made it have a cursor pointer as well. 
https://www.loom.com/share/77cf1b47f71248b99a23893b255ac044

**Review by filiptronicek — APPROVED** [review]
Thanks for the quick changes, I think PR is good to go, LGTM ✅

File: stories/2-Card.stories.js [hunk 1]
```diff
@@ -49,6 +49,20 @@ export const ContextCard = () => (
...
+    </li>
+    <li>
+      <SubtleLink className="menu-link">Logout</SubtleLink>
+    </li>
+    <li>
+      <span className="app-info">v0.01</span>
+    </li>
+  </DropdownMenuCard>
+);
+
```
Comment (bdougie) [inline stories/2-Card.stories.js:65]: Thanks so much add this to the storybook 🙏🏾

**Comment by bdougie** [comment]
I am fine to have this merge without tests, but if you were interested this could follow the pattern to test for accessibility. 

example: https://github.com/open-sauced/open-sauced/blob/main/src/tests/RepoListItem.test.js

You may see some weirdness in the test do the use of refs (those might need to be mocked).

**Review by bdougie — APPROVED** [review]
I think this LGTM as is, but left a note for the tests

File: stories/2-Card.stories.js (hunk 1 above)
Comment (blackgirlbytes) [reply stories/2-Card.stories.js:65]: Thanks! 😃

**Comment by blackgirlbytes** [comment]
> I am fine to have this merge without tests, but if you were interested this could follow the pattern to test for accessibility. 
> 
> 
> 
> example: https://github.com/open-sauced/open-sauced/blob/main/src/tests/RepoListItem.test.js
> 
> 
> 
> You may see some weirdness in the test do the use of refs (those might need to be mocked).

I'll try it out! @bdougie

**Comment by mtfoley** [comment]
Been out of pocket so this is cool to see come together 😀 @blackgirlbytes did you refer to the docs at all for your contributions and if so, did you have any thoughts on how they could be improved? I was happy to see you had included some CSS for dark mode 🤩

**Comment by blackgirlbytes** [comment]
just a note: I'll probably get a chance to add the tests over the weekend!!! Kinda busy right now.

**Comment by 0-vortex** [comment]
> I think this LGTM as is, but left a note for the tests

Tested this with `gh pr checkout 1129` and clean build. Think we're all in agreement we can push this to main and handle the minor improvements as a discussion or new feature enhancement issue.

@filiptronicek as you have brainstormed a bit(the majority) on this issue with @blackgirlbytes do you mind handling the discussion/potential feature enhancements in such a way to minimize overhead? Talent is not wasted doing chores, but then again why keep it boring :D ?

**Comment by blackgirlbytes** [comment]
> Been out of pocket so this is cool to see come together 😀 @blackgirlbytes did you refer to the docs at all for your contributions and if so, did you have any thoughts on how they could be improved? I was happy to see you had included some CSS for dark mode 🤩

@mtfoley I found the documentation easy to follow to be honest! I think before meeting bdougie..for some reason, I couldn't find the documentation (besides the ReadME), but after I found it everything seemed good. I do have a question though! How do I open an issue for writing a test for this feature? When I tried to open an issue, I could only open an issue for a bug or a feature. Would I just open issue as a feature and just edit the description?

Thanks!

**Comment by mtfoley** [comment]
@blackgirlbytes i think you’ve got the right idea! If it were me I’d use the feature template.

**Comment by 0-vortex** [comment]
> > Been out of pocket so this is cool to see come together 😀 @blackgirlbytes did you refer to the docs at all for your contributions and if so, did you have any thoughts on how they could be improved? I was happy to see you had included some CSS for dark mode 🤩
> 
> @mtfoley I found the documentation easy to follow to be honest! I think before meeting bdougie..for some reason, I couldn't find the documentation (besides the ReadME), but after I found it everything seemed good. I do have a question though! How do I open an issue for writing a test for this feature? When I tried to open an issue, I could only open an issue for a bug or a feature. Would I just open issue as a feature and just edit the description?
> 
> Thanks!

We have seen the need for the improved issue and pull request templates and are currently in the process of enhancing them with issue forms, suggestions are super welcome and useful!

**Comment by blackgirlbytes** [comment]
Gotcha!!! Thanks @0-vortex and @mtfoley !!!

**Comment by Mstandastote** [comment]
#1031

**Pull request merged**